pkg/client/condition.go

```diff
@@ -41,6 +41,11 @@ func newConditions(cos v1.ClusterOperatorStatus, targetVersion string, time meta
 			Status:             v1.ConditionUnknown,
 			LastTransitionTime: time,
 		},
+		v1.OperatorUpgradeable: {
+			Type:               v1.OperatorUpgradeable,
+			Status:             v1.ConditionUnknown,
```

Comment:
So we are fine with setting this to default as `ConditionUnknown` as this means more or less that it's possible to upgrade?

Reply:
This is just the constructor for the transient set of states in this state machine. The actually persisted states are as follows (as is asserted in the tests):

1. "In Progress" (aka initally persisted state):
"Available": "Unknown",
"Degraded": "Unknown",
"Progressing": "True",
**"Upgradeable", "False"** <- this is debatable, we could also leave this as "Unknown", but we probably don't want to upgrade during rollout.

2. "Done" (aka happy case):
"Available", "True",
"Degraded", "False",
"Progressing", "False",
**"Upgradeable", "True"**

3. "Failed" (aka unhappy case):
"Available", "False",
"Degraded", "True",
"Progressing", "False",
**"Upgradeable", "False"**